edweiss-app/app/(app)/courses/[id]/index.tsx

```diff
@@ -154,42 +145,34 @@ const CoursePage: ApplicationRoute = () => {
 				{upcomingAssignments.length > 0
 					?
 					upcomingAssignments.map((assignment) => (
-						<TView key={assignment.name} testID={testIDs.assignemtView}>
-							<AssignmentDisplay item={assignment} index={upcomingAssignments.indexOf(assignment)} isSwipeable={true} />
-						</TView>
+						<AssignmentDisplay item={assignment} index={upcomingAssignments.indexOf(assignment)} isSwipeable={true} key={assignment.name} />
```

Comment:
Your `upcomingAssignments.indexOf(assignment)` is unnecessary, right above you can add one argument (`(assignment)` into `(assignment, index)`) to grab the index of the current assignment.